### ISSUE TITLE
Can't parse specific glob patterns in mocha configurations

### ISSUE THREAD

**Issue by YPetremann** [hbenl/vscode-mocha-test-adapter#141]
I have some mocha settings in my package.json
originally I've put 
```json
{
  ...
  "mocha": {
    "spec": [
      "./!(node_modules)/**/*.{test,spec}.{mjs,js,ts,jsx,tsx}"
    ]
  },
}
```
For now I've this workaround
```json
{
  ...
  "mocha": {
    "ignore": [
      "./node_modules/**"
    ],
    "spec": [
      "./**/*.{test,spec}.{mjs,js,ts,jsx,tsx}"
    ]
  },
}
```
The result is that `./!(node_modules)/**/*.{test,spec}.{mjs,js,ts,jsx,tsx}` is only correctly handled in mocha, `./**/*.{test,spec}.{mjs,js,ts,jsx,tsx}` is handled both in mocha and mocha-test-explorer, but ignore is made only after searching files and this make a huge difference of 1s search compared to 12s search with the workaround.

**Comment by YPetremann** [comment]
I dig a bit into what's wrong and as for now:
In src/configReader.ts:342 `const fileUris = await vscode.workspace.findFiles(relativePattern, null);` use vscode algorithm and not the mocha algorithm to find files

**Comment by hbenl** [comment]
Version 2.10.0 is using the `glob` library from npm by default (just like Mocha does), so this should work now.
You can still switch back to the old implementation using `"mochaExplorer.globImplementation": "vscode"`. I'm keeping this just in case the new implementation has a bug and also because it could be slightly more performant (because VS Code has to look at all the files in the workspace anyway and probably caches some information about them).

**Comment by YPetremann** [comment]
got this
`[Error: Couldn't load options using mocha: child process exited with code 1 and signal null
	at ChildProcess.<anonymous> (c:\Users\User\.vscode\extensions\hbenl.vscode-mocha-test-adapter-2.10.0\out\optsReader.js:81:32)
	at Object.onceWrapper (events.js:422:26)
	at ChildProcess.emit (events.js:315:20)
	at maybeClose (internal/child_process.js:1021:16)
	at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)] {
  vslsStack: [ CallSite {}, CallSite {}, CallSite {}, CallSite {}, CallSite {} ]
}`

even by putting glob implementation to "vscode"

**Comment by YPetremann** [comment]
Sorry, it was a result of a partial extension update

**Comment by hbenl** [comment]
> Sorry, it was a result of a partial extension update, which is not related directly to the addon

I guess you upgraded to 2.10.0 first, which was broken due to a bug in `vsce`. I published a fixed version 2.10.1 shortly after.
